src/KubernetesClient.Models/KubernetesJson.cs

```diff
@@ -26,18 +26,49 @@ public override void Write(Utf8JsonWriter writer, TimeSpan value, JsonSerializer
 
         private sealed class KubernetesDateTimeOffsetConverter : JsonConverter<DateTimeOffset>
         {
-            private const string SerializeFormat = "yyyy'-'MM'-'dd'T'HH':'mm':'ss.ffffffK";
-            private const string Iso8601Format = "yyyy'-'MM'-'dd'T'HH':'mm':'ssK";
+            private const string RFC3339NanoFormat = "yyyy-MM-dd'T'HH':'mm':'ss.fffffffK";
+
+            private static string FormatDateTimeOffsetToSevenDigitsNanoseconds(string dateTime)
+            {
+                var isUTC = dateTime.EndsWith("Z");
```

Comment:
too complex
why not just trim extra input?

Reply:
As already pointed out by @anekdoti in https://github.com/kubernetes-client/csharp/issues/1350 the lastTransitionTimestamp should be deserializable as a Golang Time struct that follows RFC3339 where an arbitrary number of nansecond digits from [0 to 9 is valid](https://cs.opensource.google/go/go/+/refs/tags/go1.20.6:src/time/format.go;l=1494). Therefore trimming alone would not help here.
This method allows an arbitrary number of nanosecond digits from 0 to 9.